src/Domain/LeanCode.DomainModels/Model/Id.cs

```diff
@@ -0,0 +1,64 @@
+using System;
+using System.Diagnostics;
+using LeanCode.DomainModels.Serialization;
+using LeanCode.IdentityProvider;
+
+namespace LeanCode.DomainModels.Model
+{
+    [DebuggerDisplay("{Value}")]
+    [TypedIdConverter]
```

Comment:
Inb4 someone tries to (de)serialize one of these with `Newtonsoft.Json` :smile:

Reply:
Damn, MT Relay uses Newtonsoft currently. I may create a converter, but I don't think we'd like to hardcode an attribute here.

Reply:
`LeanCode.DomainModels` don't (and shouldn't) reference `Newtonsoft.Json`. We can provide the converters Json.Net converters but let's use other assembly for that.

I have the same problem with these converters - they don't really belong in the domain.

Reply:
Yeah, it might not be the best place, but the alternative is to force every user to add a proper converter factory each time they want to perform the serialization.

Reply:
This has this advantage that you will need to think about the data (and format) that you're serializing.

I somewhat agree that having this here is just easier and don't introduce any complexity (thanks to it being internal) so I'm OK with it.

Reply:
> We can provide the Json.Net converters but let's use other assembly for that.

I agree, and I wouldn't mind if they came in separate PR.